### PR TITLE
Container healthcheck

### LINKED ISSUE #107: Add health-checks to user docker containers
Add this configuration https://godoc.org/github.com/docker/docker/api/types/container#HealthConfig to https://godoc.org/github.com/docker/docker/api/types/container#Config during container creation at `lib/docker/container.go` function `CreateContainer`

Use the `curl` command for pinging the port on which the deployed application is running with a specific timeout.
`curl --fail http://localhost:PORT/ || exit 1`

### PR DESCRIPTION
**Pull request by Scar26**
Closes https://github.com/sdslabs/gasper/issues/107

### CODE REVIEW

File: lib/docker/container.go [hunk 1]
```diff
@@ -35,7 +37,15 @@ func CreateContainer(containerCfg *types.ApplicationContainer) (string, error) {
 		Volumes: map[string]struct{}{
 			volume: {},
 		},
+		Healthcheck : &container.HealthConfig{
+			Test: []string{"CMD-SHELL", fmt.Sprintf("curl --fail http://localhost:%d/ || exit 1", containerCfg.ContainerPort)},
```
Comment (alphadose) [inline lib/docker/container.go]: change `containerCfg.ContainerPort` to `containerCfg.ApplicationPort` and add the `--silent` flag to the curl command

File: lib/docker/container.go [hunk 2]
```diff
@@ -35,7 +37,15 @@ func CreateContainer(containerCfg *types.ApplicationContainer) (string, error) {
...
+			Interval: configs.ServiceConfig.Mizu.MetricsInterval*time.Second,
+			Timeout: 10*time.Second,
+			Retries: 3,
```
Comment (alphadose) [inline lib/docker/container.go:44]: Add a start period of 2 minutes to the configuration

**Review by alphadose — CHANGES_REQUESTED** [review]

File: lib/docker/container.go (hunk 2 above)
Comment (Scar26) [reply lib/docker/container.go:44]: there's no such member in the HealthConfig struct

Comment (alphadose) [reply lib/docker/container.go:44]: There is, check this out https://godoc.org/github.com/docker/docker/api/types/container#HealthConfig

File: lib/docker/container.go (hunk 1 above)
Comment (alphadose) [reply lib/docker/container.go]: @Scar26 you forgot these changes

File: lib/docker/container.go [hunk 3]
```diff
@@ -35,7 +37,15 @@ func CreateContainer(containerCfg *types.ApplicationContainer) (string, error) {
...
+		},
 	}
+
+
```
Comment (alphadose) [inline lib/docker/container.go]: Remove this extra line

**Review by alphadose — CHANGES_REQUESTED** [review]

File: lib/docker/container.go (hunk 1 above)
Comment (Scar26) [reply lib/docker/container.go]: sorry, I accidentally reverted that commit as well

File: lib/docker/container.go (hunk 3 above)
Comment (alphadose) [reply lib/docker/container.go]: @Scar26 what about this ?

**Review by alphadose — APPROVED** [review]

**Pull request merged**